### ISSUE TITLE
Mode switching doesn't work in HITL, shows multiple items ticked if it fails

### ISSUE THREAD

**Issue by LorenzMeier** [mavlink/qgroundcontrol#2525]
I'm not clear yet where things break, but once PX4 is in HITL mode mode switches are completely ignored. Might be a firmware thing, the reason I'm filing it here is a different one: The mode switch indicator still updates, resulting in two modes being selected even if the system rejected or didn't respond.

**Comment by DonLakeFlyer** [comment]
The fact that the mode switch indicator update is just a bug. That dropdown cna't have checkmarks on it. I have a fix for this. I stepped through the actual flight mode change onthe QGC side and everythign looks good from there. The change of mode is going out. The firmware just never does anything with it. That's going to need to be debugged fromthe firmware side. Also when I ran HIL with XPlane firmware spits out tons of ekf messages.

**Comment by DonLakeFlyer** [comment]
There is also a crashing bug in QGC HIL when you shutdown. I'll fix that as well.

**Comment by DonLakeFlyer** [comment]
FYI: I did get it to switch to Loiter one time. No idea what I did to make that happen. It only worked once.

**Comment by DonLakeFlyer** [comment]
I'm trying to debug the firmware side.

**Comment by DonLakeFlyer** [comment]
Here is what I found:
- If you attempt to change flight mode before you connect HIL (which turns on HIL mode) the flight mode changes are rejected by firmware. So the messages are getting to the board.
- If you attempt to changle flight mode after HIL is turned on firmware no longer responds to messages from QGC. Not on does flight mode change no longer work, but things like param refresh are not responded to either. I validated that in this case the flight mode change message is going out from the QGC side. No idea what is going on here.
